### PR TITLE
Authentication refactor

### PR DESCRIPTION
**Pull request by mongoKart**
PR changes:
- Moves each mechanism to its own page
- Adds API links to each page
- Removes Enterprise Authentication section
- Adds table on product compatibility to Auth landing page
- Small fixes mentioned in DOCSP-43931
- Other small improvements on placeholders, troubleshooting, wording

# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA:
- https://jira.mongodb.org/browse/DOCSP-40289
- https://jira.mongodb.org/browse/DOCSP-43931

### Staging Links

<li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals/authentication/aws-iam>fundamentals/authentication/aws-iam</a></li><li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals/authentication/kerberos>fundamentals/authentication/kerberos</a></li><li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals/authentication/ldap>fundamentals/authentication/ldap</a></li><li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals/authentication/oidc>fundamentals/authentication/oidc</a></li><li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals/authentication/scram>fundamentals/authentication/scram</a></li><li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals/authentication/x509>fundamentals/authentication/x509</a></li><li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/connection-troubleshooting>connection-troubleshooting</a></li><li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals>fundamentals</a></li><li><a href=https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals/authentication>fundamentals/authentication</a></li>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/fundamentals/authentication.txt [hunk 1]
```diff
@@ -17,399 +17,65 @@ Authentication Mechanisms
...
-To learn more about any of the methods or types discussed in this
-guide, see the following API Documentation:
-
-- `MongoCredential() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.MongoCredential.html>`__
-- `MongoClient() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.MongoClient.html>`__
-- `MongoClientSettings <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.MongoClientSettings.html>`__
-- `CreateCredential() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.MongoCredential.CreateCredential.html>`__
-- `CreateMongoX509Credential() <{+new-api-root+}/MongoDB.Driver/MongoDB.Driver.MongoCredential.CreateMongoX509Credential.html>`__
\ No newline at end of file
+   To learn how to establish a connection to your MongoDB deployment, see
```
Comment (rustagir) [inline source/fundamentals/authentication.txt:40]:
```suggestion
   To learn how to establish a connection to your MongoDB deployment, see the
```

**Review by rustagir — APPROVED** [review]
looks good!

**Comment by rishitb-mongodb** [comment]
@mongoKart 

- The new [SCRAM page](https://deploy-preview-284--mongodb-docs-csharp.netlify.app/fundamentals/authentication/scram/) doesn't tell the user how to use SCRAM 256 vs SCRAM 1? [Unlike the older page](https://www.mongodb.com/docs/drivers/csharp/current/fundamentals/authentication/#scram-sha-256), is this intentional? 
- I didn't see a callout for not needing a port number when connecting to Atlas. 

Everything else looks good. Thank you for the updates!

File: source/fundamentals/authentication/x509.txt [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+see the corresponding syntax:
+
+.. tabs::
+
+   .. tab:: Connection String
+      :tabid: default-connection-string
+
+      .. code-block:: csharp
+
+         var connectionString = "mongodb://<hostname>[:<port>]/?authMechanism=MONGODB-X509";
```
Comment (mongoKart) [inline source/fundamentals/authentication/x509.txt:74]: Note to tech reviewer: Rishit got a timeout exception when using this code sample to connect to his deployment. Is the code sample correct?

File: source/fundamentals/authentication/scram.txt [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+- ``<authenticationDb>``: The MongoDB database that contains the user's authentication
+  data. If you omit this parameter, the driver uses the default value, ``admin``.
+
+To use the code examples on this page, replace these placeholders with your own values.
+
+Using SCRAM Authentication in Your Application
+----------------------------------------------
+
+To use SCRAM to authenticate your MongoDB user, specify your
+MongoDB credentials, but don't specify an authentication mechanism.
```
Comment (mongoKart) [inline source/fundamentals/authentication/scram.txt:63]: Question for tech reviewer: Do we need to show how to explicitly specify SCRAM-1 or SCRAM-256? I assumed users would want to use the best one supported by their deployment.

File: source/fundamentals/authentication.txt [hunk 4]
```diff
@@ -17,399 +17,65 @@ Authentication Mechanisms
...
+
+The following table lists the authentication mechanisms supported by MongoDB and
+the {+mdb-server+} editions that each mechanism is compatible with. Click the name of
+a mechanism to learn more about how to use it with your application.
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Authentication Mechanism
```
Comment (mongoKart) [inline source/fundamentals/authentication.txt:54]: Question for tech reviewer: Is this compat table accurate?

File: source/fundamentals/authentication/x509.txt (hunk 2 above)
Comment (sanych-sun) [reply source/fundamentals/authentication/x509.txt:75]: The time-out problem related to wrong connection string (it should be `mongo+srv` for Atlas)

Comment (sanych-sun) [reply source/fundamentals/authentication/x509.txt:75]: I've provided list of additional steps to successful connection to Atlas with x509 certificate via Slack, we can discuss the code sample if you want.

File: source/fundamentals/authentication.txt (hunk 4 above)
Comment (sanych-sun) [reply source/fundamentals/authentication.txt:54]: Honestly I do not know, it's probably better to ask someone from server-side. Or @JamesKovacs might help with that.

File: source/fundamentals/authentication/scram.txt (hunk 3 above)
Comment (mongoKart) [reply source/fundamentals/authentication/scram.txt:63]: If the user omits `authMechanism` and the server doesn't support SCRAM-256, does the driver default to SCRAM-1?

I'm wondering if the user would ever need to explicitly specify SCRAM-1, or if the driver would automatically negotiate the most secure SCRAM protocol.

**Review by sanych-sun — APPROVED** [review]
LGTM

**Pull request merged**